### DISCUSSION TITLE
Perform PageBreak between two rows in the table component

### DISCUSSION THREAD

**Discussion by Puge96** [QuestPDF/QuestPDF#646] [Q&A] [answered]
Hi.

I'm generating a PDF with a table using following structure.
I need to perform a PageBreak in the middle of my table when a certain value appears in Col1 of my row data, is this possible?

```
container.PaddingTop(20).Column(c =>
{
    c.Item().Table(table =>
	{
		table.ColumnsDefinition(columns =>
		{
			columns.ConstantColumn(50);
			columns.ConstantColumn(50);
		});

		table.Header(header =>
		{
			header.cell().Text("col 1");
			header.cell().Text("col 2");
		});

		foreach (row in data.Rows) // data that stores rows
		{
			if (row.Col1 == "SomeValue")
			{
				//I have tried following but without luck.
				//table.cell().PageBreak();
				//container.PageBreak();
				//c.item().PageBreak();
			}

			table.cell().text(row.Col1);
			table.cell().text(row.Col2);
		}
	}
}
```

Thanks in advance.

**Comment by Puge96** [comment]
I've solved it by doing following:

```
container.PaddingTop(20).Column(c =>
{
    c.Item().Table(table =>
	{
		table.ColumnsDefinition(columns =>
		{
			columns.ConstantColumn(50);
			columns.ConstantColumn(50);
		});

		table.Header(header =>
		{
			header.cell().Text("col 1");
			header.cell().Text("col 2");
		});

		foreach (row in data.Rows) // data that stores rows
		{
			if (row.Col1 == "SomeValue")
			{
				// Add a new cell with a pagebreak for each column in your columns definition
				table.cell().PageBreak();
				table.cell().PageBreak();
			}
			table.cell().text(row.Col1);
			table.cell().text(row.Col2);
		}
	}
}
```

It works but I guess there is a better option than adding a fake cell with a pagebreak for each of the cells in the table.

**Reply by MarcinZiabek** [reply, marked as answer]
There are a couple of concepts that could be used. 

1) If the table is used purely as a structural element, you may consider splitting the content into separate sections within the `Column` element. This allows to manage page breaking by simply adding the `PageBreak` element to the table.

2) If keeping the entire content in the table is crucial, you can still follow your idea of inserting a specialized cell with a page break. Just ensure it covers the entire row (so not to disrupt cell ordering). You have accomplished this by inserting as many cells as there are columns. But a more straightforward approach is to use `table.Cell().ColumnSpan(2).PageBreak()`.

**Reply by Puge96** [reply]
I do need to keep the content within one table element as I don't know if I will be doing the PageBreak at all.
Utilizing ColumnSpan is way prettier than what I was doing (I have 12 columns... :-)).

Thank you very much.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
